src/api/models/users.js

```diff
@@ -20,14 +25,55 @@ var UsersSchema = new mongoose.Schema({
         default: Date.now
     },
     user_type: {
-        type: [{
-            type: String,
-            enum: ['student', 'recruiter']
-        }],
-        default: ['student']
+        type: String,
+        default: 'student',
+        validate: {
+            isAsync: true,
+            validator: function(v, cb) {
+                if (v === 'student') {
+                    cb(true);
+                } else if (v === 'recruiter') {
+                    var msg = 'recruiter must have a valid employer_id';
+                    if(/[a-f0-9]{24}/.test(this.employer_id)) {
+                        Employer.count({_id: ObjectId(this.employer_id)}, function (err, count){ 
+                            if (err) {
+                                cb(false, msg); //some kind of server error
+                            } else if (!(count > 0)) {
+                                cb(false, msg); //no employer found with this employer_id
+                            } else {
+                                cb(true);
+                            }
+                        }); 
+                    } else {
+                        cb(false, msg); //employer_id isn't a 24 character hex string
+                    }
+                } else {
+                    cb(false, 'user_type must be student or recruiter');
+                }
+            },
+            message: 'Error validating user_type'
+        },
     },
     employer_id: {
         type: String,
+        validate: {
```

Comment:
Change employer_id to be of type **Schema.types.ObjectId,**
and then use the mongoose-id-validator plugin
```
UserSchema.plugin(idvalidator);
```

Reply:
Oh and then remove the validator cause you don't need it.

Then you can probably add a required function
```
    employer_id: {
        type: Schema.Types.ObjectId,
        ref: 'Employer',
        required: function() {
            return this.user_type === 'recruiter';
        },
        default: null,
    }
```